tests/c_api_tests/test_serialization.cpp

```diff
@@ -138,6 +138,29 @@ TEST_CASE("Serialization") {
             CHECK(is_nan(node[0].u_rated));
             CHECK(source[0].id == 6);
             CHECK(source[1].id == 7);
+
```

Comment:
What is the motivation for these additions in serialization?

Reply:
whoops sorry unrelated to this PR. this is something I discussed with Laurynas and that i will probably add in another PR (there's no example for using serialization result in the C API). accidentally added to this one. Will revert